ui/admin/app/styles/app.scss

```diff
@@ -452,3 +454,103 @@
     }
   }
 }
+
+// secret editor
+.json-secret-editor {
+  .CodeMirror {
+    height: 210px;
+  }
+
+  .CodeMirror-gutters {
+    background: #31332b;
+  }
+}
+
+.secret-editor {
+  background: #272822;
+  border-radius: 4px;
+  height: 210px;
+  margin-bottom: sizing.rems(l);
+}
+
+.skeleton-wrapper {
+  color: var(--ui-gray);
+  display: flex;
+  flex-direction: row;
+  position: relative;
+}
+
+.skeleton-gutter {
+  background: #31332b;
+  border-radius: 4px 0 0 4px;
+  height: 210px;
+  width: 52px;
+}
+
+.skeleton-body {
+  width: 100%;
+
+  > div {
+    background: #31332b;
+    border-radius: 16px;
+    height: 8px;
+    margin: 10.8px;
+
+    &:first-child {
+      margin-top: 16px;
+    }
+
+    &.s {
+      width: 5%;
+    }
+
+    &.m {
+      width: 10%;
+    }
+
+    &.l {
+      width: 15%;
+    }
+
+    &.xl {
+      width: 20%;
+    }
+  }
+}
+
+.details-view {
+  font-size: 14px;
+  left: 50%;
+  margin: -12px 0 0 -165.5px;
```

Comment:
Why are these negative margins needed, is this to account for the size of the element itself? Could we have used `transform: translate(-50%, -50%)` instead?

Reply:
My preference is to avoid arbitrary pixel values like this.  Is there a way to accomplish the desired positioning in a way that doesn't depend on the element size?  If pixels are pragmatic, perhaps include some comments for context.

Reply:
Updated to use transform.